jax/_src/scipy/stats/_core.py

```diff
@@ -0,0 +1,151 @@
+    if nan_policy not in policies:
+        raise ValueError("nan_policy must be one of {%s}" %
+                         ', '.join("'%s'" % s for s in policies))
+    try:
+        # Calling jnp.sum to avoid creating a huge array into memory
+        # e.g. jnp.isnan(a).any()
+        with np.errstate(invalid='ignore'):                                               #check
+            contains_nan = jnp.isnan(jnp.sum(a))
+    except TypeError:
+        # This can happen when attempting to sum things which are not
+        # numbers (e.g. as in the function `mode`). Try an alternative method:
+        try:
+            contains_nan = jnp.nan in set(a.ravel())
+        except TypeError:
+            # Don't know what to do. Fall back to omitting nan values and
+            # issue a warning.
+            contains_nan = False
+            nan_policy = 'omit'
+            warnings.warn("The input array could not be properly "
+                          "checked for nan values. nan values "
+                          "will be ignored.", RuntimeWarning)
+
+    if contains_nan and nan_policy == 'raise':
+        raise ValueError("The input contains nan values")
+
+    return contains_nan, nan_policy
+
+def mode(a, axis=0, nan_policy='propagate'):
+
+    """
+    Return an array of the modal (most common) value in the passed array.
+    If there is more than one such value, only the smallest is returned.
+    The bin-count for the modal bins is also returned.
+    Parameters
+    ----------
+    a : array_like
+        n-dimensional array of which to find mode(s).
+    axis : int or None, optional
+        Axis along which to operate. Default is 0. If None, compute over
+        the whole array `a`.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan.
+        The following options are available (default is 'propagate'):
+          * 'propagate': returns nan
+          * 'raise': throws an error
+          * 'omit': performs the calculations ignoring nan values
+    Returns
+    -------
+    mode : ndarray
+        Array of modal values.
+    count : ndarray
+        Array of counts for each mode.
+    Examples
+    --------
+    >>> a = jnp.array([[6, 8, 3, 0],
+    ...               [3, 2, 1, 7],
+    ...               [8, 1, 8, 4],
+    ...               [5, 3, 0, 5],
+    ...               [4, 7, 5, 9]])
+    >>> from scipy import stats
+    >>> stats.mode(a)
+    ModeResult(mode=array([[3, 1, 0, 0]]), count=array([[1, 1, 1, 1]]))
+    To get mode of whole array, specify ``axis=None``:
+    >>> stats.mode(a, axis=None)
+    ModeResult(mode=array([3]), count=array([3]))
+    """
+
+    a, axis = _chk_asarray(a, axis)
+    if a.size == 0:
+        return ModeResult(jnp.array([]), jnp.array([]))
+
+    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+
+    if contains_nan and nan_policy == 'omit':
+        a = ma.masked_invalid(a)
+        return mstats_basic.mode(a, axis)
```

Comment:
We cannot fall back to numpy/scipy functions here, because they cast all inputs to regular numpy arrays, and thus are incompatible with JAX transforms. We'll have to implement these directly (or, you can punt on it now and raise `NotImplementedError`)